easybuild/easyblocks/p/petsc.py

```diff
@@ -224,20 +228,43 @@ def configure_step(self):
                         withdep = "--with-%s" % dep[1].lower()
                     self.cfg.update('configopts', '%s=1 %s-dir=%s' % (withdep, withdep, deproot))
 
+            # SCOTCH has to be treated separately since they add weird postfixes
+            # to library names from SCOTCH 7.0.1 or PETSc version 3.17.
+            if (LooseVersion(self.version) >= LooseVersion("3.17")):
+                scotch = get_software_root('SCOTCH')
+                if scotch:
+                    withdep = "--with-ptscotch"
+                    scotch_inc = [os.path.join(scotch, "include")]
+                    inc_spec = "-include=[%s]" % ','.join(scotch_inc)
+
+                    # For some reason there is a v3 suffix added to libptscotchparmetis
+                    # which is the reason for this new code.
+                    req_scotch_libs = ['libptesmumps.a', 'libptscotchparmetisv3.a',
+                                       'libptscotch.a', 'libptscotcherr.a', 'libesmumps.a',
+                                       'libscotch.a', 'libscotcherr.a']
```

Comment:
please sort these library names alphabetically, much easier for human eyes?

Reply:
Done!. :)